src/hotspot/share/opto/escape.cpp

```diff
@@ -463,8 +463,15 @@ bool ConnectionGraph::compute_escape() {
 // if at least one scalar replaceable allocation participates in the merge.
 bool ConnectionGraph::can_reduce_phi_check_inputs(PhiNode* ophi) const {
   bool found_sr_allocate = false;
-
+  int nof_input_phi_nodes = 0;
   for (uint i = 1; i < ophi->req(); i++) {
+    if (ophi->in(i)->is_Phi()) {
+      // ignore phi node with more than one input phi nodes
+      if (++nof_input_phi_nodes > 1) {
+        NOT_PRODUCT(if (TraceReduceAllocationMerges && !found_sr_allocate) tty->print_cr("Cannot reduce Phi %d. More than one input phi node.", ophi->_idx);)
```

Comment:
remove condition !found_sr_allocate

Reply:
Are you still planning to remove this condition?